### DISCUSSION TITLE
ESPHome Version 2024.7.0 BREAKING CHANGE - Microwakeword

### DISCUSSION THREAD

**Discussion by BigBobbas** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#43] [Announcements]
Hi guys,
the ESPHome release today 2024.7.0 has a breaking change that will prevent compile due to an update to microwakeword2.
this is a 'breaking change' detailed in the release notes / change log https://esphome.io/changelog/2024.7.0 . The new format for the config can also be seen here https://esphome.io/components/micro_wake_word.html
I am working on a new config with other features and will also include the fix for this. I'm expecting to make this available later today. and will post an announcement once available.

If you want to fix the issue prior to the update, you can change the microwakeword config to the following:
```yaml
micro_wake_word:
  models: 
    - model: ${micro_wake_word_model}
```
Cheers 
and as ever , thank you for your support

Rob

**Comment by solarjunkev** [comment]
Hi Rob 
I have just installed yr fantastic software for this ESPhome device 
I have an ESP32-S3-Box-3B
I have the latest ESPHome software loaded 2024.7.0 (I can see this setting under the ESPHome device in HA)
This is what I see:
Firmware: 2024.7.0 (Jul 20 2024, 08:14:56)
Hardware: 2024.06.01
Does this mean I have yr latest firmware? 
I can trigger local voice via holding the buttom R button but I still can't trigger the microwake word without holding this button
I have set Wyoming to listen as the local wakeword etc 
It's like the device is "not listening" as it sits idle
What am I doing wrong?
What do I check?
How do I check for your latest firmware for the SP-Box3 and how do I upload it? thanks  
Thanks

**Reply by BigBobbas** [reply]
hi, 
to use microwakeword you need to first set wakeword to 'on device' from the info page.  The default wake word is then 'ok nabu'

The latest stable version will be the one that you have installed. There is also a experimental version with extra features and the details can be found [HERE](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/discussions/44>) To install first copy and paste the current contents of the device card in the ESPHome dashboard to notepad, or any text editor so that you have a copy as a backup. Then you need to copy the contents of the config file from the link above  and paste it into the device card in the ESPHome dashboard overwriting anything that is currently in there. Click save and then install. If you get an error at the end of compiling, close the card editor. click on the 3 dot menu on the card and then select 'clean build files' then try the install again. 

Cheers

**Reply by solarjunkev** [reply]
Cheers mate. I'd rather wait for stable versions. Your next one includes the fix right?.
The one I have clearly doesn't work due to the break. 
I am a newbie to ESPHome
Where do I get _your_ (specific) firmware upgrades? ...when you release them.. (Say with this new fix)?
(edit : ah ok so I download the code, unzip...)
And what is the firmware upgrade process?
(edit - I have to reload the firmware and upload it via USB - like the original flash? )
Sorry for nube q- the upgrade process is not in the manual?
Cheers

**Reply by BigBobbas** [reply]
the fix is already applied. It would only effect the installation not the wakeword  working. If it isn't working then it is something else that is causing it.
When the updates are added they will be added as an announcement with links to the files. All that is required is to either flash again from the web installer, or to just copy and paste the contents of the file as described above.

I have just tested and it looks like there may be an issue when switching from homeassistant to 'on device' 

I would suggest in this case trying https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/dev/s3b_dev.yaml

**Reply by solarjunkev** [reply]
Ok cheers for that.
Sorry for the newb questions. Like I said I am new to ESPHome<>HA
Will we ever see an easier update process?
In other words when you release your new firmware we see "update available" and can simply select "update" in the ESPHome HA dashboard? 
How come this isn't available  now? 
Thanks!

**Comment by BigBobbas** [comment]
Hi , 
yes I am hoping to implement an easier update process in the future. This is a relatively new project and there is only myself writing the configs. It is quite a long process to get everything working in the way it does and to include all of the features etc without issues and errors. 
I am not a coder or developer and have no previous experience so things are taking a while for me to figure out.
But I will get there eventually ;)

Cheers 👍

**Reply by solarjunkev** [reply]
No worries! 
(I didn't realise the ESPHome/HA integration made that firmware upgrade process so difficult - that's all..)
Nobody actually "tells anyone" the "reasons" in this HA journey....
It just "is"
😊
